File: katas/8 kyu/beginner-series-number-1-school-paperwork/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def paperwork(n, m):
  return n*m if n>= 0 and m >= 0 else 0
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def paperwork(n, m):
    return max(n, 0)*max(m, 0)
# ----------------------------------- TEST ----------------------------------- #
@test.describe("Fixed Tests")
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(paperwork(5,5), 25, "Failed at Paperwork(5,5)")
        test.assert_equals(paperwork(-5,5), 0, "Failed at Paperwork(-5,5)")
        test.assert_equals(paperwork(5,-5), 0, "Failed at Paperwork(5,-5)")
        test.assert_equals(paperwork(-5,-5), 0, "Failed at Paperwork(-5,-5)")
        test.assert_equals(paperwork(5,0), 0, "Failed at Paperwork(5,0)")

print(-5 * 2)

File: katas/8 kyu/beginner-series-number-1-school-paperwork/solution.js
// beginner-series-number-1-school-paperwork
/* ------------------------------- MY SOLUTION ------------------------------ */
function paperwork(n, m) {
  if (n > 0 && m > 0) {
    return n*m
  }
  else return 0
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function paperwork(n, m) {
  return n > 0 && m > 0 ? n * m : 0
}
/* ---------------------------------- TEST ---------------------------------- */
const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Simple Tests", function(){
  it("Testing for fixed tests", () => {
    assert.strictEqual(paperwork(5,5), 25, 'Failed at paperwork(5, 5)');
    assert.strictEqual(paperwork(5,-5), 0, 'Failed at paperwork(5, -5)');
    assert.strictEqual(paperwork(-5,-5), 0, 'Failed at paperwork(-5, -5)');
    assert.strictEqual(paperwork(-5,5), 0, 'Failed at paperwork(-5, 5)');
    assert.strictEqual(paperwork(5,0), 0, 'Failed at paperwork(5, 0)');
  });
});
